### PR TITLE
Gas injectors are now buildable

### LINKED ISSUE #15789: air injector item and consistency issues
### What is the problem?

air injectors can't be built; to work around this you can connect pipes to them, however there does not seem to be a way to activate them and no gas is injected through them, causing me a ton of problems with my painstakingly constructed burn chamber.
### Why is it a problem, or what should have happened?

i should be able to build the things with my RPD or metal as an engiborg, and they should function when connecting with standard piping
### How can the problem be reproduced?

connect straight pipes to the air injector with arbitrary pressure; observe nothing happening

note: i connected the mix loop all the way from atmos to the misc testing lab's burn chamber to see if i could get some ridiculously high temperature. while building up my H/E loop so i could use it to heat canisters i realized air injectors were not possible to build which seems a little weird. to get around this i simply connected my pipe of infinite plasma at 4000 KPA to the air injector but nothing happened.

i was pretty miffed because the whole thing took the whole round RIP my dream ;_;

anyway these should be standard atmos equipment and they should be buildable and work; as is they are kind of snowflake and lack consistency. durandan said they are admin spawn only and i'm not sure why.

### PR DESCRIPTION
**Pull request by Core0verload**
You can now construct/deconstruct gas injectors. They can also be controlled manually, just like volumetric pumps.

Fixes #15789. Also fixes a bug with "MAX" button not working on volumetric pumps.

:cl: CoreOverload
add: Gas injectors are now buildable!
/:cl:

### CODE REVIEW

File: code/game/machinery/pipe/construction.dm [hunk 1]
```diff
@@ -120,6 +122,7 @@ GLOBAL_LIST_INIT(pipeID2State, list(
 		"[PIPE_CONNECTOR]" 		= "connector", \
 		"[PIPE_UVENT]" 			= "vent", \
 		"[PIPE_SCRUBBER]" 		= "scrubber", \
+		"[PIPE_INJECTOR]"		= "injector", \
```
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: holy shit pipe construction is so bad

File: code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm [hunk 2]
```diff
@@ -139,3 +144,53 @@
...
+
+/obj/machinery/atmospherics/components/unary/outlet_injector/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, \
+																		datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "atmos_pump", name, 310, 115, master_ui, state)
+		ui.open()
+
+/obj/machinery/atmospherics/components/unary/outlet_injector/ui_data()
+	var/data = list()
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm]: just do `. = list()` and then use `.` throughout and remove the explicit return

**Comment by duncathan** [comment]
also the freeze PR isn't closed yet so

File: code/game/machinery/pipe/construction.dm (hunk 1 above)
Comment (duncathan) [inline code/game/machinery/pipe/construction.dm]: your code is fine tho

**Comment by Core0verload** [comment]
Someone said "the freeze will be over in two days" two days ago, so I figured it's the time.

File: code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm (hunk 2 above)
Comment (Core0verload) [inline code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm]: Any reason to do this?

**Comment by WJohn** [comment]
Freeze is over.

File: code/game/machinery/pipe/construction.dm (hunk 1 above)
Comment (lzimann) [inline code/game/machinery/pipe/construction.dm]: Yea, that is what stopped me from doing this exact pr like an year ago.

File: code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm (hunk 2 above)
Comment (lzimann) [inline code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm]: Doesn't really matter, using `data` is cleaner to read than `.`

Comment (duncathan) [inline code/modules/atmospherics/machinery/components/unary_devices/outlet_injector.dm]: i forgot that pipe construction is not resource sensitive

**Pull request merged**